deploy/generate_addon.sh

```diff
@@ -52,20 +52,52 @@ create_rolebinding() {
 }
 
 get_secret_name_from_service_account() {
-  echo -e "\\nGetting secret of service account ${SERVICE_ACCOUNT_NAME} on ${NAMESPACE}"
-  SECRET_NAME=$(kubectl get sa "${SERVICE_ACCOUNT_NAME}" --namespace="${NAMESPACE}" -o jsonpath="{.secrets[].name}")
+  SERVER_MINOR_VERSION=$(kubectl version -ojson | jq -r '.serverVersion.minor')
```

Comment:
a major + minor validation is preferred.

Reply:
if major version >1, we still use the new logic 
do we need a check for major version == 0?

Reply:
`major version >= 1` is also fine, thanks.

Reply:
updated